### PR TITLE
docs(v2): Added switch config docs to theme-classic API

### PR DESCRIPTION
**Pull request by Drewbi**
## Motivation

The new color mode options were added to the docusaurus-theme-classic api page but did not include the switchConfig options.

### Have you read the [Contributing Guidelines on pull requests](https://github.com/facebook/docusaurus/blob/master/CONTRIBUTING.md#pull-requests)?

Yup

## New documentation

Screen Shot 2020-08-03 at 7 07 47 pm

### CODE REVIEW

File: yarn.lock [hunk 1]
```diff
@@ -8346,58 +8346,6 @@ doctrine@^3.0.0:
   dependencies:
     esutils "^2.0.2"
 
-docusaurus@^2.0.0-alpha.61:
```
Comment (slorber) [inline yarn.lock:8349]: hi,

why is this needed in this PR?

File: website/docs/api/docusaurus.config.js.md [hunk 2]
```diff
@@ -174,9 +174,7 @@ module.exports = {
       switchConfig: {
         darkIcon: '🌙',
         lightIcon: '\u2600',
-        // React inline style objects
-        // see https://reactjs.org/docs/dom-elements.html#style
-        darkIconStyle: {
+        darkIconStyle: { // CSS object with camelCased properties
```
Comment (slorber) [inline website/docs/api/docusaurus.config.js.md:177]: why not leave my comment? It's not worth to document camelCased etc, React developers know about inline style objects, and non-react devs will read the link.

File: website/docs/theme-classic.md [hunk 3]
```diff
@@ -33,6 +33,26 @@ module.exports = {
...
       // using user system preferences, instead of the hardcoded defaultMode
       respectPrefersColorScheme: false,
+
+      // Dark/light switch icon options
+      switchConfig: {
+        // Icon for the switch while in dark mode
+        darkIcon: '🌙',
+
+        // CSS to apply to dark icon with camelCase properties,
+        // More information available here: https://reactjs.org/docs/dom-elements.html#style
```
Comment (slorber) [inline website/docs/theme-classic.md:43]:
```suggestion
        // React inline style objects
        // see https://reactjs.org/docs/dom-elements.html#style
```

File: yarn.lock (hunk 1 above)
Comment (slorber) [reply yarn.lock:8349]: will be fixed in https://github.com/facebook/docusaurus/pull/3197

Comment (Drewbi) [reply yarn.lock:8349]: Ah yep, I'll remove

File: website/docs/api/docusaurus.config.js.md (hunk 2 above)
Comment (Drewbi) [reply website/docs/api/docusaurus.config.js.md:177]: Sorry, my thinking was if we were adding more explanation to the theme classic api page we could reduce the comments in the docusaurus.config.js api doc as I didn't see any other explanatory comments there but you're right, the previous version was better.

File: website/docs/theme-classic.md [hunk 4]
```diff
@@ -33,6 +33,27 @@ module.exports = {
...
+        // CSS to apply to dark icon,
+        // React inline style object
+        // see https://reactjs.org/docs/dom-elements.html#style
+        darkIconStyle: {
+          marginLeft: '2px',
+        },
+
+        // Unicode icons such as `\u2600` will work if
+        // the code is 4 characters long and not 5 e.g. `\u1F602`
+        lightIcon: '\u2600',
```
Comment (slorber) [inline website/docs/theme-classic.md:51]:
```suggestion
        // Unicode icons such as `\u2600` will work if
        // the code is 4 characters long and not 5 e.g. `\u1F602`
        lightIcon: '\u2600',
        
```
```suggestion
        // Unicode icons such as '\u2600' will work
        // Unicode with 5 digits require brackets: '\u{1F602}'
        lightIcon: '\u{1F602}'
        
```

TIL :)

File: website/docs/theme-classic.md [hunk 5]
```diff
@@ -33,6 +33,28 @@ module.exports = {
...
+        // React inline style object
+        // see https://reactjs.org/docs/dom-elements.html#style
+        darkIconStyle: {
+          marginLeft: '2px',
+        },
+
+        // Unicode icons such as '\u2600' will work
+        // Unicode with 5 digits require brackets: '\u{1F602}'
+        lightIcon: '\u{1F602}'
+
```
Comment (slorber) [inline website/docs/theme-classic.md:52]:
```suggestion
```

**Comment by slorber** [comment]
LGTM

File: website/docs/theme-classic.md [hunk 6]
```diff
@@ -33,6 +33,27 @@ module.exports = {
...
+        // CSS to apply to dark icon,
+        // React inline style object
+        // see https://reactjs.org/docs/dom-elements.html#style
+        darkIconStyle: {
+          marginLeft: '2px',
+        },
+
+        // Unicode icons such as '\u2600' will work
+        // Unicode with 5 digits require brackets: '\u{1F602}'
```
Comment (slorber) [inline website/docs/theme-classic.md:50]:
```suggestion
        // Unicode with 5 chars require brackets: '\u{1F602}'
```
